### PR TITLE
fix: only render debug artifacts when info is available

### LINKED ISSUE #25319: [IATR](M1.0) - Update artifact links on Debug page to only show when urls available
*Current Behavior*
Artifact Links shown in `DebugFailedTest` and `GroupedDebugFailedTest` are always present, even if the test does not have URLs for the different artifacts present.

Example from `DebugFailedTest.vue`
```
debugArtifacts: [
      { icon: 'TERMINAL_LOG', text: 'View Log', url: runInstance?.stdoutUrl! },
      { icon: 'IMAGE_SCREENSHOT', text: 'View Screenshot', url: runInstance?.screenshotsUrl! },
      { icon: 'PLAY', text: 'View Video', url: runInstance?.videoUrl! },
    ],
```

*Desired Behavior*
Code should check to see if the various artifacts are present by checking the three boolean fields that map to each artifact type and only rendering the buttons if they are true.  The fields are `hasStdout`, `hasVideo` and `hasScreenshots`

### PR DESCRIPTION
**Pull request by ZachJW34**
- Closes #25319

### User facing changelog
n/a - Feature branch

### Additional details
Only render debug artifacts if the corresponding `has<attr>` is true and the `<attr>Url` is defined.

### Steps to test

1. Check out this branch and the cypress-services feature branch, make sure the cypress-services dev is running
2. Create a project if you don't have one
3. Choose a repo with Cypress setup and a simple test that will fail
4. Add `video: false` to the cypress config
5. Run `CYPRESS_INTERNAL_ENV=development npx cypress run --record --key <RECORD_KEY>`
6. Edit `DebugContainer.vue` and change `runByNumber(runNumber: <RUN_NUMBER>)`, the last step should have recorded a run to your locally running dashboard and you can find the run number there

You can disable screenshots if you like and record another run.

### How has the user experience changed?

https://user-images.githubusercontent.com/25158820/210644704-ebc0c237-1285-48c9-8fdb-09e2e0f37ef9.mov

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/app/src/debug/DebugFailedTest.cy.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import DebugFailedTest from './DebugFailedTest.vue'
-import type { TestResults } from './DebugSpec.vue'
```
Comment (ZachJW34) [inline packages/app/src/debug/DebugFailedTest.cy.tsx]: This showed as `any` when importing types from a vue file into a tsx file so I moved them to another file to get better type safety.

File: packages/app/vite-env.d.ts [hunk 2]
```diff
@@ -2,3 +2,9 @@
 /// <reference types="vite-plugin-pages/client" />
 /// <reference types="vite-plugin-vue-layouts/client" />
 /// <reference types="vite-svg-loader" />
+
+declare module '*.vue' {
```
Comment (ZachJW34) [inline packages/app/vite-env.d.ts]: This stops VS Code from complaining about Vue component imports in `tsx` files

File: packages/app/src/debug/DebugFailedTest.cy.tsx (hunk 1 above)
Comment (warrensplayer) [reply packages/app/src/debug/DebugFailedTest.cy.tsx]: I am not opposed to moving the types to their own file in this instance when they are used multiple places.  There does seem to be a "right" way to export things from a Vue component  that uses the composition api.  You would use a separate normal `<script>` tag.  See here: https://vuejs.org/api/sfc-script-setup.html#usage-alongside-normal-script

Comment (ZachJW34) [reply packages/app/src/debug/DebugFailedTest.cy.tsx]: I'm going to revert these changes, I gave @astone123 suggestion a go following the docs for [enabling Volar take-over mode](https://vuejs.org/guide/typescript/overview.html#volar-takeover-mode) and my IDE is a lot happier.

File: packages/app/src/debug/DebugContainer.vue [hunk 3]
```diff
@@ -52,7 +52,7 @@
 
 <script setup lang="ts">
 import { gql } from '@urql/vue'
-import { computed } from '@vue/reactivity'
```
Comment (ZachJW34) [inline packages/app/src/debug/DebugContainer.vue:57]: Saw this in a few files, thought I'd fix them all

File: packages/app/src/debug/DebugFailedTest.cy.tsx [hunk 4]
```diff
@@ -199,4 +182,37 @@ describe('<DebugFailedTest/>', () => {
 
     cy.percySnapshot()
   })
+
+  it('conditionally renders artifacts', () => {
+    const render = (testResult: TestResults) => cy.mount(() =>
+      <DebugFailedTest failedTestsResult={[testResult]} groups={[group1]} expandable={false}/>)
```
Comment (warrensplayer) [inline packages/app/src/debug/DebugFailedTest.cy.tsx]:
```suggestion
    const render = (testResult: TestResults) => cy.mount(() =>
      (<div data-cy="test-group">
        <DebugFailedTest failedTestsResult={[testResult]} groups={[group1]} expandable={false}/>
      </div>))
```
Only one minor suggestion which is to wrap the component with the same data-cy attribute to allow the styling to show the buttons on hover.

**Review by warrensplayer — APPROVED** [review]
Approving. Just added one suggestion.

File: packages/app/src/debug/utils/debugArtifacts.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+export type ArtifactType = 'TERMINAL_LOG' | 'IMAGE_SCREENSHOT' | 'PLAY'
+
+export type DebugArtifact = { icon: ArtifactType, text: string, url: string }
+
+export const getDebugArtifacts = (instance: CloudRunInstance | null): DebugArtifact[] => {
+  const debugArtifacts: DebugArtifact[] = []
+
+  if (instance?.hasStdout && instance.stdoutUrl) {
+    debugArtifacts.push({ icon: 'TERMINAL_LOG', text: 'View Log', url: instance.stdoutUrl })
```
Comment (mike-plummer) [inline packages/app/src/debug/utils/debugArtifacts.ts]: Should "View log" and other `text` field values be i18n?

**Review by mike-plummer — APPROVED** [review]

**Comment by ZachJW34** [comment]
Going to merge, CI failed due to a flaky test and I've created an issue + PR for skipping it: https://github.com/cypress-io/cypress/pull/25378

**Pull request merged**